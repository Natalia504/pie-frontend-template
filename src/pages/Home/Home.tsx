import React from "react";
import { Counter } from "@pie/components";
import { Layout } from "../../components/Layout";
import { useSelector, useDispatch } from "react-redux";

import { countSelector, counter } from "../../store/counter";
import { Action } from "redux";

const {
  actions: { increment, decrement }
} = counter;

export const Home: React.FC = () => {
  const count = useSelector(countSelector);
  const dispatch = useDispatch();
  return (
    <Layout>
      <Counter
        value={count}
        onIncrement={(): Action => dispatch(increment())}
        onDecrement={(): Action => dispatch(decrement())}
      />
    </Layout>
  );
};
