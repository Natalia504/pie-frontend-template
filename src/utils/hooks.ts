import { useEffect } from "react";

export const useMount = (
  mountFunction: () => void,
  cleanupFunction?: () => void
): void => {
  useEffect(() => {
    mountFunction();
    return cleanupFunction && cleanupFunction();
  }, []); /* eslint-disable-line react-hooks/exhaustive-deps */
};
