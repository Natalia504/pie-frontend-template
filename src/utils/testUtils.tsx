import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import { ThemeProvider, createMuiTheme } from "@material-ui/core";
import { pieLightTheme } from "@pie/components";

const mockState = {
  counter: 0
};

export const mockStore = configureMockStore()(mockState);

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult =>
  render(
    <Provider store={mockStore}>
      <ThemeProvider theme={createMuiTheme(pieLightTheme)}>{ui}</ThemeProvider>
    </Provider>
  );
