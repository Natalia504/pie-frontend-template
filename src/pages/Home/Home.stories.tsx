import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";
import { withA11y } from "@storybook/addon-a11y";
import { Provider } from "react-redux";

import { Home } from "./Home";
import { mockStore } from "../../utils/testUtils";

storiesOf("Home", module)
  .addDecorator(withA11y)
  .addDecorator(withKnobs)
  .add("Kitchen Sink", () => (
    <Provider store={mockStore}>
      <Home />
    </Provider>
  ));
