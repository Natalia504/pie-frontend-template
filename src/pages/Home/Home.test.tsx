import React from "react";
import { RenderResult } from "@testing-library/react";

import { Home } from "./Home";
import { renderWithProviders } from "../../utils/testUtils";

describe("Layout tests", () => {
  const setup = (): RenderResult => renderWithProviders(<Home />);
  it("renders", () => {
    const { baseElement } = setup();
    expect(baseElement).toMatchSnapshot();
  });
});
