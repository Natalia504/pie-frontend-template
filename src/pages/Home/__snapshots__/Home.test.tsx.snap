// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests renders 1`] = `
<body>
  <div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        App Header
      </p>
      <div
        class="MuiFormControl-root makeStyles-root-31"
      >
        <div
          class="MuiBox-root MuiBox-root-43 makeStyles-box-32 makeStyles-box-36"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root makeStyles-button-33 makeStyles-button-37 MuiIconButton-sizeSmall"
            tabindex="0"
            title="subtract"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 13H6c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1z"
                />
              </svg>
            </span>
          </button>
          <div
            class="MuiInputBase-root MuiInput-root makeStyles-input-34 MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              pattern="/\\\\d+/"
              size="2"
              type="number"
              value="0"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root makeStyles-button-33 makeStyles-button-37 MuiIconButton-sizeSmall"
            tabindex="0"
            title="add"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
                />
              </svg>
            </span>
          </button>
        </div>
        <p
          class="MuiFormHelperText-root makeStyles-formHelperText-35 MuiFormHelperText-filled"
        />
      </div>
    </div>
  </div>
</body>
`;
